notify-send.scm

```diff
@@ -0,0 +1,177 @@
+                         (let ((wrapped-callback
+                                (lambda (n a d)
+                                  (callback n a d)
+                                  (g-main-loop-quit loop))))
+                           (notification-add-action notification
+                                                    action-name
+                                                    label
+                                                    wrapped-callback
+                                                    user-data
+                                                    free-func)))))
+                    actions)
+          ;; Show the notification and run the main loop
+          (notification-show notification)
+          (g-main-loop-run loop)
+          (notify-uninit))
+        ;; If not waiting, show notification and uninit immediately
+        (begin
+          (notification-show notification)
+          (notify-uninit)))))
+
+;; XXX: Copied from (gnu services configuration).
+(define (list-of pred?)
+    (lambda (x)
+      (if (list? x)
+          (every pred? x)
+          #f)))
+
+;; XXX: Copied from (rde features predicates).
+(define (maybe pred?)
+  (lambda (x)
+    (or (pred? x) (not x))))
+
+;; XXX: Copied from (rde features).
+(define-syntax ensure-pred
+  (syntax-rules ()
+    ((ensure-pred pred field)
+     (when (not (pred field))
+       (raise (condition
+               (&message
+                (message
+                 (format
+                  #f "~a: The predicate '~a' is not satisfied with value '~a'."
+                  'field
+                  (procedure-name pred)
+                  field)))))))))
+
+;; ensure-pred requires procedure-name to be defined
+(define (urgency? x)
+  (memq x '(low normal critical)))
+
+(define (hint? x)
+  (and (pair? x)
+       (memq (car x) '(action-icons
+                       category
+                       desktop-entry
+                       image-data
+                       image-path
+                       resident
+                       sound-file
+                       sound-name
+                       suppress-sound
+                       transient
+                       x
+                       y
+                       urgency))))
+
+(define (action? x)
+  (and (list? x)
+       (match x
+         ((name label callback user-data free-func)
+          (and (ensure-pred string?    name)
+               (ensure-pred string?    label)
+               (ensure-pred procedure? callback)
+               (ensure-pred string?    user-data)
+               (ensure-pred procedure? free-func)))
+         (_ #f))))
+
+(define maybe-string?    (maybe string?))
+(define maybe-urgency?   (maybe urgency?))
+(define maybe-integer?   (maybe integer?))
+(define maybe-symbol?    (maybe symbol?))
+(define list-of-hints?    (list-of hint?))
+(define list-of-actions?  (list-of action?))
+(define maybe-procedure? (maybe procedure?))
+
+(define* (notify-send* summary #:key
```

Comment:
I think I prefer to go simple and add only the `notify-send` function. IMO this is too much code for not a lot of benefit.
I would just let the user be wrong. This library is simple, and that's important for me.